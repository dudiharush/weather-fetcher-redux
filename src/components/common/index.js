export { Button } from './button/Button';
export { SearchBox } from './searchBox/SearchBox';
