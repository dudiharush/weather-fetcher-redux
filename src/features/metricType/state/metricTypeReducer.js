import { SET_TEMPERATURE_UNIT } from './constants';
import { units } from 'app/services/openWeatherMap/units';

const initialState = units.celsius;

export const metricType = (state = initialState, action) => {
    if(action.type === SET_TEMPERATURE_UNIT){
      return action.tempType;
    }

  return state;
};
