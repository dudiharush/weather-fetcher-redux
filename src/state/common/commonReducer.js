export const commonReducer = (state, actions, action) => {
  if (actions.includes(action.type)) {
    const { ...actionPayload } = action;

    return {
      ...state,
      ...actionPayload
    };
  }

  return state;
};
