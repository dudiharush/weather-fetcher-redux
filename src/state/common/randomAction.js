import { RANDOM_ACTION_TYPE } from '../constants';

export const randomAction = () => ({
  type: RANDOM_ACTION_TYPE
});
