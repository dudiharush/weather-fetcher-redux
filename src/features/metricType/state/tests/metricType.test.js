import { randomAction } from 'app/state/common/randomAction';
import { metricType } from '../metricTypeReducer';

describe('metricType test suite', () => {
  describe('default', () => {
    it('should return metric type when not match action is passed', () => {
      const action = randomAction();

      expect(metricType(undefined, action)).toMatchSnapshot();
    });
  });
});
