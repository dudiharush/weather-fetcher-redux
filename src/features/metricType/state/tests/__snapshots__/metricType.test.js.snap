// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metricType test suite default should return metric type when not match action is passed 1`] = `"metric"`;
